types/src/system/auction/mod.rs

```diff
@@ -152,14 +159,17 @@ pub trait Auction:
         Ok(updated_amount)
     }
 
-    /// For a non-founder validator, implements essentially the same logic as add_bid, but reducing
-    /// the number of tokens and calling unbond in lieu of bond.
+    /// For a non-founder validator, this entry point implements essentially the same logic as
```

Comment:
I don't think we should refer to `add_bid` here since it leaves too many open questions that I think the doc comment should answer.

For example,
* what happens if `amount` is more than they have in their bid account?
* what happens if they pass `amount` set to `0`? (It's not the same as for `add_bid` by the looks of it - we don't error here)
* if the bid account is left with `0` is the validator's entry in the `Key::Bid` section of the global state removed or left with a value of `0`?